test/helpers/mock_server.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require 'webrick'
+
+module MockServer
+  extend self
+
+  def start(port)
+    tid = Thread.new do
+      TestServer.new(hostname: '0.0.0.0', port: port).start
+    end
+    poll_until_ready(hostname: '0.0.0.0', port: port)
+    puts "Created test server #{tid}, port: #{port}"
+    tid
+  end
+
+  def cleanup(tid)
+    Thread.kill(tid)
+    puts "Killed test server #{tid}."
+  end
+
+  def poll_until_ready(hostname:, port:, time_to_wait: 1)
+    start_time = Time.now.to_i
+    begin
+      TCPSocket.new(hostname, port).close
+    rescue Errno::ECONNREFUSED, Errno::ECONNRESET
+      if Time.now.to_i > start_time + time_to_wait
+        raise "Couldn't reach the service on hostname #{hostname} port #{port} after #{time_to_wait}s"
+      else
+        retry
+      end
+    end
+  end
+end
+
+class TestServer
+  def initialize(hostname:, port:)
+    dev_null = WEBrick::Log.new("/dev/null", 7)
+    @server = WEBrick::HTTPServer.new(
+      Port:  port,
+      BindAddress: hostname,
+      Logger: dev_null,
+      AccessLog: dev_null,
+    )
+    @server.mount '/', Handler
+  end
+
+  def start
+    @server.start
+  end
+end
+
+class Handler < WEBrick::HTTPServlet::AbstractServlet
+  def do_GET(request, response) # rubocop:disable all
+    status, content_type, body = process(request)
+    response.status = status
+    response['Content-Type'] = content_type
+    response.body = body
+  end
+
+  def process(env)
+    response_code = env.path.delete("/")
+    response_code = '200' if response_code == ""
+    [response_code, 'text/html', 'Success']
```

Comment:
nit: I know this is copied from previous code, but technically just "Success" isn't a valid `text/html` document, at least not in modern HTML. Is just `text/plain` sufficient?